pkg/utils/kubernetes/kubernetes.go

```diff
@@ -199,3 +199,16 @@ func GetLoadBalancerIngress(ctx context.Context, client client.Client, namespace
 
 	return "", errors.New("`.status.loadBalancer.ingress[]` has an element which does neither contain `.ip` nor `.hostname`")
 }
+
+// IsEndpointsReconciledByKubeControllerManager checks if a given Endpoints object is reconciled by kube-controller-manager.
+// It looks at the corresponding Service (same name and namespace) and checks whether a selector is specified or not.
+func IsEndpointsReconciledByKubeControllerManager(ep corev1.Endpoints, services []corev1.Service) bool {
+	for _, service := range services {
```

Comment:
This might be very costly if there are many endpoints/services in the cluster. What about creating a map[string]*corev1.Service that maps `namespace/name` to a service object?

Reply:
Good point, I adapted the logic to use a simple string set for storing, which Endpoints objects should be ignored. PTAL :)